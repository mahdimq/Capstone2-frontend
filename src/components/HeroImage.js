import React from 'react';
import styled from 'styled-components';

const StyledHero = styled.div`
background:
	linear-gradient(
    to bottom,
      rgba(0, 0, 0, 0.1) 39%,
      rgba(0, 0, 0, 0.2) 41%,
      rgba(0, 0, 0, 0.8) 100%
    ),
    url(${(props) => props.image}), #1c1c1c;

	background-size: 100%, cover;
	background-position: center, center;
	width: 100%;
	height: 600px;
	position: relative;
	animation: animateImage 1s;

	.heroimage-content {
		max-width: 1280px;
		padding: 1.5em;
		margin: 0 auto;
  }

  .heroimage-text {
    z-index: 100;
    max-width: 700px;
    position: absolute;
    bottom: 40px;
    margin-right: 1.5em;
    min-height: 100px;
    color: #fff;

    h1 {
      font-family: sans-serif;
      font-size: 3em;
      color: #fff;

      @media screen and (max-width: 720px) {
        font-size: 38px;
        color: #fff;
      }
    }

    p {
      font-family: san-serif;
      font-size: 22px;
      line-height: 26px;
      color: #fff;

      @media screen and (max-width: 720px) {
        font-sizeL 1em;
        line-height: 20px;
        color: #fff;
      }
    }

    @media screen and (max-width: 720px) {
      max-width: 100$
    }
  }

  @keyframes animateImage {
    from {
      opacity: 0
    }
    to {
      opacity: 1
    }
  }
`;

function HeroImage({ image, title, text }) {
	return (
		<StyledHero image={image}>
			<div className='heroimage-content'>
				<div className='heroimage-text'>
					<h1>{title}</h1>
					<p>{text}</p>
				</div>
			</div>
		</StyledHero>
	);
}

export default HeroImage;
