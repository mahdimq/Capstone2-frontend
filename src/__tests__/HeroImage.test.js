import React from 'react';
import { render } from '@testing-library/react';
import HeroImage from '../components/HeroImage';

it('renders without crashing', function () {
	render(<HeroImage />);
});

it('matches snapshot', function () {
	const { asFragment } = render(<HeroImage />);
	expect(asFragment()).toMatchSnapshot();
});
