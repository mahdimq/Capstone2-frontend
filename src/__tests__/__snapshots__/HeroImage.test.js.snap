// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ kRpwiH"
  >
    <div
      class="heroimage-content"
    >
      <div
        class="heroimage-text"
      >
        <h1 />
        <p />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
